pkg/service/auth_context.go

```diff
@@ -23,44 +27,125 @@ type AuthContext struct {
 	Authorization map[*config.AuthorizationConfig]interface{}
 }
 
-// Evaluate evaluates all steps of the auth pipeline (identity → metadata → policy enforcement)
-func (self *AuthContext) Evaluate() error {
-	self.Identity = make(map[*config.IdentityConfig]interface{})
-	self.Metadata = make(map[*config.MetadataConfig]interface{})
-	self.Authorization = make(map[*config.AuthorizationConfig]interface{})
-
-	// identity (authentication)
-	identityConfigs := self.API.IdentityConfigs
-	for i := range identityConfigs {
-		c := identityConfigs[i]
-		if ret, err := c.Call(self); err != nil {
+// AuthObjectConfig provides an interface for APIConfig objects that implements a Call method
+type AuthObjectConfig interface {
```

Comment:
I was thinking... Maybe instead of "object" we could name these something that helps one to understand better the purpose, like "AuthConfigExecutor" or "AuthConfigEvaluator" or (simpler) "AuthEvalutor".

WDYT?

Reply:
You're right, it makes better sense along the changes you propose renaming to "evaluators" instead of "getters"